cobc/cobc.c

```diff
@@ -8648,6 +8648,18 @@ set_cobc_defaults (void)
 		 *p == '1')) {
 		cb_unix_lf = 1;
 	}
+	p = cobc_getenv ("COBC_DIAGNOSTICS_SHOW_CARET");
```

Comment:
I want to argue to keep it: most of the options in the compiler do not need an env variable because they are project-dependent, i.e. the project developers should whether these options should be set or not for their project. On the contrary, the way errors are displayed is user-dependent, not project-dependent. As such, users may want to set or unset these options in their config files, so that they are used by `make` automatically. So the env variable is needed for that (unless we add a user config file for gnucobol).

Reply:
I see the point - but we shouldn't do this for a single option, a user may want to have the diagnostic name or not, he maybe even don't care about any warnings, ...
A project specific setup that handles this would set default flags in the makefile and passes additional flags from the user's environment or include file (depending on the setup either before or after the default makefile flags).

Actually the way projects can nicely be setup via Makefiles is the main reason for not having a global env var or a config file (other than dialect options). There's a FR about this on SF if you want to give that more thoughts - for this PR, please drop that.